src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -71,9 +71,25 @@ object LightGBMUtils {
   def getNumExecutors(dataset: Dataset[_], defaultListenPort: Int): Int = {
     val executors = getExecutors(dataset, defaultListenPort)
     if (executors.isEmpty) {
-      // Case when run locally in local[*]
-      val blockManager = BlockManagerUtils.getBlockManager(dataset)
-      blockManager.master.getMemoryStatus.size
+      val master = dataset.sparkSession.sparkContext.master
+      if (master == "local") 1
+      else {
+        val capture = "local\\[([*\\d+])\\]".r
```

Comment:
1. If `capture` here refers to the fact that the regexp is capturing,
   then it's a bad name, similar to using "variable" in a variable name.
   Same for `numCoresCapture` below.

2. The regexp seems broken: I think that what you want to use is

       "local\\[(\\*|\\d+)\\]".r

3. Also, there's some weird thing below about splitting
   `numCoresCapture` which seems wrong: I don't see any commas that
   could appear there.

Reply:
Ah, good point, I think the regex can just be:
"local\\[(*)\\]".r
The '*' part can be any of:

local : Run Spark locally with one worker thread (i.e. no parallelism at all).

local[K] : Run Spark locally with K worker threads (ideally, set this to the number of cores on your machine).

local[K,F] : Run Spark locally with K worker threads and F maxFailures (see spark.task.maxFailures for an explanation of this variable)

local[*] : Run Spark locally with as many worker threads as logical cores on your machine.

local[*,F] : Run Spark locally with as many worker threads as logical cores on your machine and F maxFailures.

Reference:
https://jaceklaskowski.gitbooks.io/mastering-apache-spark/content/spark-local.html
https://stackoverflow.com/questions/32356143/what-does-local-mean-in-spark

Reply:
Otherwise, it looks like this might work, I'll test it out (and then I won't need to do the split by comma):
"local\\[(\\*|\\d+)(:?,\\d+)?\\]".r